File: RGPV.c
#include<stdio.h>
int mark[100];
int count=0;

//Function to add reasult
void add(){
    printf("Enter mark:");
    scanf("%d",&mark[count]);
    count++;
    printf("Result Added Successfull!!\n");
}
//Function to show result
void show(){
    if(count==0){
        printf("No data found\n");
        return;
    }
    printf("\nstudent marks:\n");
    for(int i=0;i<count;i++){
        printf("student %d :%d\n",i+1,mark[i]);
    }
}
//Function to calculate average
void aver(){
    int sum=0;
    if(count==0){
        printf("No data found\n");
        return;
    }
    for(int i=0;i<count;i++){
        sum+=mark[i];
    }
    float ave=(float)sum/count;
    printf("Averge mark: %f\n",ave);

}
//Function to find Highest Number
void high(){
    if(count==0){
        printf("No data found\n");
        return;
    }
    int max=mark[0];
    for(int i=1;i<count;i++){
        if(mark[i]>max)
        max=mark[i];
    }
    printf("Highest mark=%d\n",max);
}
int main(){
    int choice;
    for(;;){
         printf("\n===== RGPV Result Analyzer =====\n");
         printf("\n1.Add result\n2.Show Result\n3.Average Mark\n4.Highest Number\n5.Exit\n");
         printf("Enter Choice:");
         scanf("%d",&choice);
         switch(choice){  
         case 1:
         add();
         break;
         case 2:
         show();
         break;
         case 3:
         aver();
         break;
         case 4:
          high();
          break;
          case 5:
          printf("\nExiting......");
          return 0;
          default:
          printf("Invalid input");
        }


    }
    return 0;
}

File: RGPV.py
mark=[]
def add():              #Function to add mark
    marks=int(input("Enter Student mark:"))
    mark.append(marks)
    print("Mark added Successfully!!")

def show():            #Function to show all result
    if len(mark)==0:
        print("No data found")
        return
    print("\nStudent marks:")
    for i,m in enumerate(mark,start=1):
        print(f"students{i}:{m}")

def ave():             #Function to calculate average mark
     if len(mark)==0:
        print("No data found")
        return
     av=sum(mark)/len(mark)
     print(f"Average mark={av:.2f}")


def high():           #Function to find highest Number
     if len(mark)==0:
        print("No data found")
        return
     top=max(mark)
     print(f"Highest mark={top}")

while True:
    print("\n.....RGPV Result Analyzer.....")
    print("1:Add mark\n2:Show mark\n3:Average mark\n4:Highest mark\n5:Exit")
    choice= input("Enter your choice:")
    if(choice=="1"):
        add()
    elif(choice=="2"):
        show()
    elif(choice=="3"):
        ave()
    elif(choice=="4"):
        high()
    elif(choice=="5"):
        print("Exiting.....")
        break
    else:
        print("Invalid Input")